### PR TITLE
Fix lookup of job id from pods

### PR DESCRIPTION
**Pull request by gabrielhartmann**

### CODE REVIEW

**Review by rostyboost — APPROVED** [review]

File: titus_isolate/model/kubernetes_workload.py [hunk 1]
```diff
@@ -63,20 +63,16 @@ def get_job_annotations_from_pod(pod: V1Pod) -> dict:
...
         job_id = metadata.labels.get(LABEL_KEY_JOB_ID, None)
+
     if job_id is None:
         # legacy, very few pods launched a while ago
-        job_id = metadata.annotations.get(ANNOTATION_KEY_JOB_ID, 'UNKNOWN_JOB_ID')
+        job_id = metadata.annotations.get(ANNOTATION_KEY_JOB_ID, None)
+
+    if job_id is None:
+        log.error("failed to extract job ID for pod: %s", pod.metadata.name)
+        job_id = 'UNKNOWN_JOB_ID'
```
Comment (solarkennedy) [inline titus_isolate/model/kubernetes_workload.py:75]: I think this function should either raise or return none, and make the caller deal with it, not return this string.

File: titus_isolate/model/kubernetes_workload.py [hunk 2]
```diff
@@ -63,20 +63,16 @@ def get_job_annotations_from_pod(pod: V1Pod) -> dict:
...
 
     return {
         JOB_ID_KEY: job_id,
```
Comment (solarkennedy) [inline titus_isolate/model/kubernetes_workload.py:78]: Do we need the complexity of returning annotations now? 
Can this function just be `get_job_id_from_pod` ?

File: tests/utils.py [hunk 3]
```diff
@@ -226,8 +225,6 @@ def get_simple_test_pod(v1=False) -> V1Pod:
     pod = test_pod_json()
     if v1:
         annotations = pod['metadata']['annotations']
-        annotations[ANNOTATION_KEY_POD_SPEC_VERSION] = "1"
-        annotations[V1_ANNOTATION_KEY_JOB_ID] = annotations[ANNOTATION_KEY_JOB_ID]
 
         del annotations[ANNOTATION_KEY_JOB_ID]
```
Comment (solarkennedy) [inline tests/utils.py:229]: This does not reflect reality. A v1 pod *does* have this annotation

**Review by solarkennedy — APPROVED** [review]

File: tests/utils.py (hunk 3 above)
Comment (gabrielhartmann) [reply tests/utils.py:229]: I don't understand what's in a v1 pod and what isn't anymore.  We really only care about the _label_ now.  What would you like to see?

File: titus_isolate/model/kubernetes_workload.py (hunk 1 above)
Comment (gabrielhartmann) [reply titus_isolate/model/kubernetes_workload.py]: good idea, fixed

File: titus_isolate/model/kubernetes_workload.py (hunk 2 above)
Comment (gabrielhartmann) [reply titus_isolate/model/kubernetes_workload.py]: yes

File: tests/utils.py (hunk 3 above)
Comment (solarkennedy) [reply tests/utils.py:229]: I would like to see both of these lines gone (you already got line 230)
``` 
         annotations[V1_ANNOTATION_KEY_JOB_ID] = annotations[ANNOTATION_KEY_JOB_ID]

        del annotations[ANNOTATION_KEY_JOB_ID]
```

File: titus_isolate/model/utils.py [hunk 4]
```diff
@@ -36,7 +36,11 @@ def get_workload(identifier) -> Optional[KubernetesWorkload]:
         pod = pod_manager.get_pod(identifier)
         if pod is not None:
             log.info("Got pod from kubernetes: %s", identifier)
-            return KubernetesWorkload(pod)
+            try:
+                return KubernetesWorkload(pod)
+            except Exception as e:
```
Comment (solarkennedy) [inline titus_isolate/model/utils.py:41]: Cool. This will be a lot easier to troubleshoot the next time

**Pull request merged**